torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -507,43 +507,57 @@ void SchedulerRuntimeInfo::collectVectorizationInfo(
 size_t SchedulerRuntimeInfo::collectMaxVectorizeSize(
     const at::Tensor& tensor,
     size_t max_vector_size_in_byte) {
+  const size_t max_vector_size = max_vector_size_in_byte / tensor.itemsize();
+
+  // If all dimensions are size 1 with stride 1, do not impose any
+  // restriction on vector size.
+  const bool all_dim_size_one = std::all_of(
+                                    tensor.sizes().begin(),
+                                    tensor.sizes().end(),
+                                    [](const auto size) {
+                                      return size == 1;
+                                    }) &&
+      std::all_of(tensor.strides().begin(),
+                  tensor.strides().end(),
+                  [](const auto stride) { return stride == 1; });
+
+  if (all_dim_size_one) {
+    return max_vector_size;
+  }
+
+  // If the innermost stride is not 1, no vectorization is allowed.
+  if (tensor.stride(-1) != 1) {
```

Comment:
What if inner most dimension is broadcast, then it doesn't matter what the stride claims to be?

Reply:
I agree. I'd rather keep the behavior unchanged in this PR, so I'll leave a note to look into it as a separate PR.

Reply:
Updated to ignore broadcast domains.